nengo/tests/conftest.py

```diff
@@ -178,6 +189,10 @@ def pytest_addoption(parser):
 
 
 def pytest_runtest_setup(item):
+    if (item.config.getvalue('compare') and
+            not getattr(item.obj, 'compare', None)):
```

Comment:
I don't think that block is needed at all. It was intended for filtering comparison and non-comparison tests, but that is already happening in `pytest_collection_modifyitems`.